src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int ARRAY_DOUBLING = 2;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(getHash(key), key, value, null);
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (currentNode.key == key || key != null && key.equals(currentNode.key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(getHash(key), key, value, null);
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        if (++size > threshold) {
```

Comment:
Make this check in the beginning, also take into account that in case you already have such a key, you do not need to increase the size